core/ir/opnd_api.h

```diff
@@ -2424,6 +2425,15 @@ INSTR_INLINE
 opnd_t
 opnd_create_reg(reg_id_t r);
 
+DR_API
+INSTR_INLINE
+/**
+ * Returns a copy of a register operand with the reg number increased
+ * by \p amount.
```

Comment:
Please add that this really only makes sense for register sets with numbered and ordered operands and only when increasing the number within one set of names.  I.e., this is a little non-sensical for x86 RCX etc. type registers, or for say adding 5 to the top-numbered register.